### PR TITLE
ddl: return error for canceled job

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/50052

Problem Summary:
- when enable tidb_enable_fast_create_table, if a job is cancelled, the user can't see the job error msg.
### What changed and how does it work?
- return job.Error when job is canceled
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/tests/fail/fail_db_test.go
Comment (D3Hunter) [inline pkg/ddl/tests/fail/fail_db_test.go:347]:
```suggestion
	testfailpoint.Enable(t, "github.com/pingcap/tidb/pkg/ddl/mockPanicInRunDDLJob", `1*panic("panic test")`))
```

File: pkg/ddl/tests/fail/fail_db_test.go
Comment (D3Hunter) [inline pkg/ddl/tests/fail/fail_db_test.go:342]:
```suggestion
```

**Review by D3Hunter — APPROVED** [review]

File: pkg/ddl/ddl_worker.go
Comment (tangenta) [inline pkg/ddl/ddl_worker.go:1066]:
```suggestion
		// return job.Error to let caller know the job is cancelled.
```

**Review by wjhuang2016 — COMMENTED** [review]
The title leading is not correct, should be package name.

File: pkg/ddl/tests/fail/fail_db_test.go
Comment (D3Hunter) [inline pkg/ddl/tests/fail/fail_db_test.go:346]: we can delete this line

**Review by tangenta — APPROVED** [review]

**Comment by D3Hunter** [comment]
/hold

let wait final decision after discuss

**Pull request merged**
